### ISSUE TITLE
[BUG] Long file lock on workspace folder after closing Visual Studio Code

### ISSUE THREAD

**Issue by ocallesp** [microsoft/vscode-dotnettools#1095]
### Describe the Issue

After closing a Visual Studio Code workspace which contained a .NET solution, the workspace remains being used by some running processes, often for over a minute or two. Those processes are the following:

dotnet.exe using <repo folder>\.git\objects\pack\pack-\<some numbers\>.idx

Repro Steps:

1. Open VSCode
2. In VSCode clone git repository
![image](https://github.com/microsoft/vscode-dotnettools/assets/510598/a4d1b3a9-b18d-4305-871c-332f064b1ce9)
3. Clone https://github.com/dotnet/project-system.git
4. Close VSCode
5. Try to delete the folder of project-system
6. Open Process Explorer in Windows and go to Find Handle and search for the folder project-system

![image](https://github.com/microsoft/vscode-dotnettools/assets/510598/0e6713e8-8a68-4156-aeef-f6397750e16d)

**Comment by ocallesp** [comment]
I found this issue while trying to debug this other bug https://github.com/microsoft/vscode-dotnettools/issues/716

**Comment by lifengl** [comment]
I wonder whether any msbuild process could be the reason as the current working directory. After closing VS or workspace all CDK/project system processes would be killed immediately, only msbuild nodes would keep alive so they could be reused in a future session. MsBuild nodes are started by msbuild internally/

**Comment by webreidi** [comment]
@nagilson any thoughts on this one?

**Comment by nagilson** [comment]
Nope, I don't think this is related to anything the install tool is doing. Im not super familiar with CPS, tbh.
